### ISSUE TITLE
Calendar: onChange event is not fired with pattern="dd.MM.yyyy"

### ISSUE THREAD

**Issue by Zerounodue** [primefaces/primefaces#3054]
##  Issue: p:calendar
onChange event is not fired if pattern is "dd.MM.yyyy"

If the pattern is "dd.MM.yyyy HH:mm" the event gets fired

## 1) Environment
- PrimeFaces version: 6.1.0
- Does it work on the newest released PrimeFaces version? no
- Does it work on the newest sources in GitHub? no
- Application server + version: WildFly 10
- Affected browsers: Tested on Google Chrome

## 2) Expected behavior

onChange() event is fired after selecting or changing manually a date

## 3) Actual behavior
onChange() is not fired

## 4) Steps to reproduce

1: Click on da calendar input

2: selecta a dare
## 5) Sample XHTML

<p:calendar
	onchange="fillParentInput(this)"
	value="#{myBean.myDate}"
	locale="de"
	navigator="true"
	pattern="dd.MM.yyyy"
	yearRange="2014:2099" />

## 6) Sample bean

	public Date getMyDate() {
		return myDate;
	}

**Comment by jxmai** [comment]
I believe this ticket is somewhat related to https://github.com/primefaces/primefaces/issues/2811. The regression source may still be #1371 .  It seems like the `onchange` has broken in multiple ways during the timepicker 1.6.3 upgrade.

To replicate:

```xhtml
<html xmlns="http://www.w3.org/1999/xhtml"
	xmlns:h="http://java.sun.com/jsf/html"
	xmlns:f="http://java.sun.com/jsf/core"
	xmlns:p="http://primefaces.org/ui">

<h:head>
</h:head>
<h:body>
	<h:form>
		<!-- <p:calendar onchange="alert('hello')" value="#{testBean.date}"
			pattern="dd.MM.yyyy HH:mm" /> -->
		<p:calendar onchange="alert('hello')" value="#{testBean.date}" />
	</h:form>
</h:body>
</html>
```

```java
package test;

import java.io.Serializable;
import java.util.Date;

import javax.faces.bean.ManagedBean;
import javax.faces.bean.ViewScoped;

@ManagedBean(name="testBean")
@ViewScoped
public class TestBean implements Serializable{

	/**
	 * 
	 */
	private static final long serialVersionUID = -9110143420906726458L;
	
	private Date date;

	public Date getDate() {
		return date;
	}

	public void setDate(Date date) {
		this.date = date;
	}

}

```

**The onchange callback will never fire**

----

> If the pattern is "dd.MM.yyyy HH:mm" the event gets fired

But if we change the date pattern to `dd.MM.yyyy HH:mm`


```xhtml
<html xmlns="http://www.w3.org/1999/xhtml"
	xmlns:h="http://java.sun.com/jsf/html"
	xmlns:f="http://java.sun.com/jsf/core"
	xmlns:p="http://primefaces.org/ui">

<h:head>
</h:head>
<h:body>
	<h:form>
		<p:calendar onchange="alert('hello')" value="#{testBean.date}"
			pattern="dd.MM.yyyy HH:mm" />
		<!-- <p:calendar onchange="alert('hello')" value="#{testBean.date}" /> -->
	</h:form>
</h:body>
</html>
```

**The callback will fire twice**

![snapshot1](https://user-images.githubusercontent.com/10467831/33750872-76104dd4-db9c-11e7-8279-e05efb0d5239.png)

**Comment by huseyinturkmenoglu** [comment]
I tried this.  There isn't an issue,

You could try pattern is "dd.MM.yyyy" in the PrimeFaces version 6.3.0. And it will work.

**Comment by jxmai** [comment]
@HuseyinTurkmenoglu Are you sure ? (as your post leads to a ticket closure). I can surely show you that this issue has not been resolved.

Please build the Lastest PF code as of today: up to https://github.com/primefaces/primefaces/commit/e748a61e5eb2366c6639db13f2056409104aa5f6

LOG:

```
INFO: Running on PrimeFaces 6.3-SNAPSHOT
```

> You could try pattern is "dd.MM.yyyy" in the PrimeFaces version 6.3.0. And it will work.


and this:


```html
<html xmlns="http://www.w3.org/1999/xhtml"
	xmlns:h="http://java.sun.com/jsf/html"
	xmlns:f="http://java.sun.com/jsf/core"
	xmlns:p="http://primefaces.org/ui">

<h:head>
</h:head>
<h:body>
	<h:form>
		<p:calendar onchange="alert('hello')" value="#{testBean.date}"
			pattern="dd.MM.yyyy" />
		<!-- <p:calendar onchange="alert('hello')" value="#{testBean.date}" /> -->
	</h:form>
</h:body>
</html>
```

**and using timepicker only but not changing the date directly, there is no alert at all**

![image](https://user-images.githubusercontent.com/10467831/38833901-6989938a-418c-11e8-8170-40e8dca79f3e.png)


**If you use the pattern, `dd.MM.yyyy HH:mm`, as in the description. callback will fire twice (on Date pick and on Date menu close).**


First one:

![image](https://user-images.githubusercontent.com/10467831/38834034-d766d534-418c-11e8-9ef2-2b538bb88042.png)


Second one:

![image](https://user-images.githubusercontent.com/10467831/38834052-e9b7ebe2-418c-11e8-89ea-7ff233d24c7c.png)

**Comment by huseyinturkmenoglu** [comment]
PrimeFaces use a DataPicker in the p:calendar, you can use the onselect instead of onchange. 

For information: 
[https://stackoverflow.com/questions/6471959/jquery-datepicker-onchange-event-issue](url)

**Comment by Zerounodue** [comment]
onchange and onselect are obviously not the same event. In my case I need the onchange event to be fired

**Comment by Gyurmatag** [comment]
I'm facing the same problem. @Zerounodue have you got any solution for this?

**Comment by eliakocher** [comment]
@Gyurmatag  I have found no solution to this problem.

**Comment by fr60** [comment]
Same problem here ! no solution. using different pattern will result in ajax dateSelect event not fire

**Comment by melloware** [comment]
@fr60 @eliakocher @Gyurmatag If you need a quick fix to this right now just add this script somewhere on your page or in your JS file.
```javascript
PrimeFaces.widget.Calendar.prototype.fireDateSelectEvent = function() {
    this.input.trigger('change');
    this.callBehavior('dateSelect');
};
```

**Comment by melloware** [comment]
This is fixed in DatePicker in 9.0 which is the recommended Calendar to use.
